gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -136,6 +145,92 @@ func (vm *VMKeeper) getGnoStore(ctx sdk.Context) gno.Store {
 	}
 }
 
+// Namespace can be either a user or crypto address.
+// XXX: Uppercase should be valid but transform into lowercase ?
+var reNamespace = regexp.MustCompile(`^gno.land/(?:r|p)/([a-zA-Z]+[_a-zA-Z0-9]+)`)
+
+// checkNamespacePermission check if the user as given has correct permssion to on the given pkg path
+func (vm *VMKeeper) checkNamespacePermission(ctx sdk.Context, creator crypto.Address, pkgPath string) error {
+	const sysUsersPkg = "gno.land/r/demo/users"
+
+	store := vm.getGnoStore(ctx)
+
+	// if `sysUsersPkg` does not exists -> skip validation.
+	usersPkg := store.GetPackage(sysUsersPkg, false)
+	if usersPkg == nil {
+		return nil
```

Comment:
Let's add a warning reminding that the check is skipped until someone (anyone!) uploads a contract there.

In other words, I can DDoS the chain with a single transaction.

Reply:
Do we want to enforce the namespaces on each gno chain, or just on Gno.land?

Reply:
Other chains do as they please regarding this point.